File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
        int count =0, v= 0;
       for(int x =0; x<= nums.size() -1 ; x++){

            if(count ==0) v = nums[x];
            
           if ( v == nums[x]) count++;

            else count--;

        }

        return v;

        
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
         
         int number = nums[0];

         int count =0;

         for(int i =0; i< nums.length ; i++){

             if(count == 0){
                number = nums[i];
             } 
            if(nums[i] == number ){
                count++;
            } else {
                count = count -1;
                 
            }
         }

         return number;
    }
}